### PR TITLE
Added refine and bound options for mail command

### PR DESCRIPTION
**Pull request by Skormie**
* **Addressed Issue(s)**: 
A small quality of life change for scripters.

* **Server Mode**: 
Both

* **Description of Pull Request**: 
Added refine option for mail script command and also made some slight corrections to the script_commands document.

### CODE REVIEW

**Comment by rAthenaAPI** [comment]
This pull request **introduces 1 alert** when merging 2ed1212d5494284590b331b533d2f9b50150fa75 into 3dd388ac66580e72f56826d142ef69d2525188b7 - [view on LGTM.com](https://lgtm.com/projects/g/rathena/rathena/rev/pr-07665532f2e5d36fae457b07626efad4a7a9bc05)

**new alerts:**

* 1 for Comparison result is always the same

---

This project has automated code review enabled, but doesn't use the <a href="https://github.com/apps/lgtm-com">LGTM GitHub App</a>. Migrate over by <a href="https://github.com/apps/lgtm-com/installations/new">installing the app</a>. Read about the <a href="https://lgtm.com/blog/announcing_github_apps_migration">benefits of migrating to GitHub Apps</a> in the blog.

---

*Comment posted by [LGTM.com](https://lgtm.com)*

File: src/map/script.cpp [hunk 1]
```diff
@@ -24180,12 +24181,42 @@ BUILDIN_FUNC(mail){
...
+		data = script_getdata(st, 9);
+
+		if (!mail_sub(st, data, sd, 9, &name, &start, &end, &id)) {
+			return SCRIPT_CMD_FAILURE;
+		}
+
+		for (i = 0; i < num_items && start < end; i++, start++) {
+			struct item_data* item = itemdb_exists(msg.item[i].nameid);
+
+			ref = (int32)__64BPRTSIZE(get_val2(st, reference_uid(id, start), reference_getref(data)));
```
Comment (aleos89) [inline src/map/script.cpp]: The declaration for `ref` could be moved down inside `if (!script_hasdata(st, 9)) {` or you could just rid of it and directly assign it to `msg.item[i].refine`.

File: src/map/script.cpp [hunk 2]
```diff
@@ -24180,12 +24181,42 @@ BUILDIN_FUNC(mail){
...
+
+			script_removetop(st, -1, 0);
+
+			if (item->type == IT_WEAPON || item->type == IT_ARMOR || item->type == IT_SHADOWGEAR) {
```
Comment (aleos89) [inline src/map/script.cpp]: You might want to add a check to make sure the item is able to be refined. IE:
```suggestion
			if (!item->flag.no_refine && (item->type == IT_WEAPON || item->type == IT_ARMOR || item->type == IT_SHADOWGEAR)) {
```

File: src/map/script.cpp [hunk 3]
```diff
@@ -24180,12 +24181,42 @@ BUILDIN_FUNC(mail){
...
+				if (ref > MAX_REFINE)
+					ref = MAX_REFINE;
+			}
+			else
+				ref = 0;
+			if (ref < 0)
```
Comment (aleos89) [inline src/map/script.cpp]: This check is useless since `ref` is an `unsigned int`.

**Pull request merged**